tools/copy_bom/src/util.rs

```diff
@@ -48,6 +68,140 @@ pub fn calculate_file_hash(filename: &str) -> String {
     hash
 }
 
+/// find dependent shared objects for this file
+/// Currently, we only support dependent shared objects with absolute path.
+pub fn find_dependent_shared_objects(file_path: &str) -> HashSet<(String, String)> {
+    let mut shared_objects = HashSet::new();
+    // find dependencies for the input file
+    // first, we find the dynamic loader for the elf file, if we can't find the loader, return empty shared objects
+    let dynamic_loader = auto_dynamic_loader(file_path);
+    if dynamic_loader.is_none() {
+        return shared_objects;
+    }
+    let (dynamic_loader_src, dynamic_loader_dest) = dynamic_loader.unwrap();
+    shared_objects.insert((dynamic_loader_src.clone(), dynamic_loader_dest));
+    let output = output_of_dynamic_loader(&file_path, &dynamic_loader_src);
+    if let Ok(output) = output {
+        let mut objects = extract_dependencies_from_output(&file_path, output);
+        for item in objects.drain() {
+            shared_objects.insert(item);
+        }
+    }
+    shared_objects
+}
+
+/// get the output of the given dynamic loader.
+fn output_of_dynamic_loader(
```

Comment:
Is there any meaningful name than output? "output" can be anything, so actually, it means nothing.

Reply:
fixed